crates/common/tedge_config/src/tedge_toml/tedge_config/mqtt_config.rs

```diff
@@ -142,34 +153,30 @@ impl TEdgeConfig {
         Ok(mqtt_config)
     }
 
-    /// Returns an authentication configuration for an MQTT client that will connect to the Cumulocity MQTT broker.
+    /// Returns an authentication configuration for an MQTT client that will connect to the MQTT
+    /// broker of a cloud given in the parameter.
     pub fn mqtt_auth_config_cloud_broker(
         &self,
-        c8y_profile: Option<&str>,
+        cloud: &dyn CloudConfig,
     ) -> anyhow::Result<MqttAuthConfigCloudBroker> {
-        let c8y = self.c8y.try_get(c8y_profile)?;
-
-        let mut mqtt_auth = MqttAuthConfigCloudBroker {
-            ca_dir: Some(c8y.root_cert_path.clone().into()),
-            ca_file: None,
-            client: None,
-        };
+        let ca_path = cloud.root_cert_path().to_path_buf();
```

Comment:
Does this still work when the user is using a directory path for the `<cloud>.root_cert_path`? As it seems that later on the `ca_path` variable is being assigned to the `cert_file` property of the `MqttAuthClientConfigCloudBroker` which seems to suggest that it would break something.

Reply:
In this particular method the only use of `ca_path` I see is in the final return, but I've inlined the binding for clarity in 0848bf1a